inc/sitemaps/class-sitemaps.php

```diff
@@ -0,0 +1,432 @@
+	private $current_page = 1;
+
+	/** @var WPSEO_Sitemap_Timezone $timezone */
+	private $timezone;
+
+	/** @var WPSEO_Sitemaps_Router $router */
+	public $router;
+
+	/** @var WPSEO_Sitemaps_Renderer $renderer  */
+	public $renderer;
+
+	/** @var WPSEO_Sitemap_Provider[] $providers */
+	public $providers;
+
+	/**
+	 * Class constructor
+	 */
+	public function __construct() {
+
+		add_action( 'after_setup_theme', array( $this, 'reduce_query_load' ), 99 );
+		add_action( 'pre_get_posts', array( $this, 'redirect' ), 1 );
+		add_action( 'wpseo_hit_sitemap_index', array( $this, 'hit_sitemap_index' ) );
+
+		$options           = WPSEO_Options::get_all();
+		$this->max_entries = $options['entries-per-page'];
+		$this->timezone    = new WPSEO_Sitemap_Timezone();
+		$this->router      = new WPSEO_Sitemaps_Router();
+		$this->renderer    = new WPSEO_Sitemaps_Renderer();
+		$this->providers   = array( // TODO API for add/remove. R.
+			new WPSEO_Post_Type_Sitemap_Provider(),
+			new WPSEO_Taxonomy_Sitemap_Provider(),
+			new WPSEO_Author_Sitemap_Provider(),
+		);
+
+		if ( ! empty( $_SERVER['SERVER_PROTOCOL'] ) ) {
+			$this->http_protocol = sanitize_text_field( $_SERVER['SERVER_PROTOCOL'] );
+		}
+	}
+
+	/**
+	 * Check the current request URI, if we can determine it's probably an XML sitemap, kill loading the widgets
+	 */
+	public function reduce_query_load() {
+
+		if ( ! isset( $_SERVER['REQUEST_URI'] ) ) {
+			return;
+		}
+
+		$request_uri = $_SERVER['REQUEST_URI'];
+		$extension   = substr( $request_uri, -4 );
+
+		if ( false !== stripos( $request_uri, 'sitemap' ) && in_array( $extension, array( '.xml', '.xsl' ) ) ) {
+			remove_all_actions( 'widgets_init' );
+		}
+	}
+
+	/**
+	 * Register your own sitemap. Call this during 'init'.
+	 *
+	 * @param string   $name     The name of the sitemap.
+	 * @param callback $function Function to build your sitemap.
+	 * @param string   $rewrite  Optional. Regular expression to match your sitemap with.
+	 */
+	public function register_sitemap( $name, $function, $rewrite = '' ) {
+		add_action( 'wpseo_do_sitemap_' . $name, $function );
+		if ( ! empty( $rewrite ) ) {
+			add_rewrite_rule( $rewrite, 'index.php?sitemap=' . $name, 'top' );
+		}
+	}
+
+	/**
+	 * Register your own XSL file. Call this during 'init'.
+	 *
+	 * @param string   $name     The name of the XSL file.
+	 * @param callback $function Function to build your XSL file.
+	 * @param string   $rewrite  Optional. Regular expression to match your sitemap with.
+	 */
+	public function register_xsl( $name, $function, $rewrite = '' ) {
+		add_action( 'wpseo_xsl_' . $name, $function );
+		if ( ! empty( $rewrite ) ) {
+			add_rewrite_rule( $rewrite, 'index.php?xsl=' . $name, 'top' );
+		}
+	}
+
+	/**
+	 * Set the sitemap current page to allow creating partial sitemaps with wp-cli
+	 * in a one-off process.
+	 *
+	 * @param integer $current_page The part that should be generated.
+	 */
+	public function set_n( $current_page ) {
```

Comment:
Can't rename because public API (internal property name changes for clarity wherever possible). I don't think this is even used in own code, if I remember right it was provided for third party needs to implement something.